pop-api/src/v0/fungibles.rs

```diff
@@ -75,354 +75,357 @@ pub mod events {
-	pub fn create(id: AssetId, admin: AccountId, min_balance: Balance) -> Result<()> {
+	pub fn create(id: TokenId, admin: AccountId, min_balance: Balance) -> Result<()> {
 		build_dispatch(CREATE)
-			.input::<(AssetId, AccountId, Balance)>()
+			.input::<(TokenId, AccountId, Balance)>()
 			.output::<Result<()>, true>()
 			.handle_error_code::<StatusCode>()
 			.call(&(id, admin, min_balance))
 	}
 
-	/// Start the process of destroying a token with a given asset ID.
+	/// Start the process of destroying a token.
 	///
 	/// # Parameters
-	/// - `id` - The ID of the asset.
+	/// - `token` - The token to be destroyed.
 	#[inline]
-	pub fn start_destroy(id: AssetId) -> Result<()> {
+	pub fn start_destroy(token: TokenId) -> Result<()> {
 		build_dispatch(START_DESTROY)
-			.input::<AssetId>()
+			.input::<TokenId>()
 			.output::<Result<()>, true>()
 			.handle_error_code::<StatusCode>()
-			.call(&(id))
+			.call(&(token))
 	}
 
-	/// Set the metadata for a token with a given asset ID.
+	/// Set the metadata for a token.
 	///
 	/// # Parameters
-	/// - `id`: The identifier of the asset to update.
-	/// - `name`: The user friendly name of this asset. Limited in length by `StringLimit`.
-	/// - `symbol`: The exchange symbol for this asset. Limited in length by `StringLimit`.
-	/// - `decimals`: The number of decimals this asset uses to represent one unit.
+	/// - `token`: The token to update.
+	/// - `name`: The user friendly name of this token.
+	/// - `symbol`: The exchange symbol for this token.
+	/// - `decimals`: The number of decimals this token uses to represent one unit.
 	#[inline]
-	pub fn set_metadata(id: AssetId, name: Vec<u8>, symbol: Vec<u8>, decimals: u8) -> Result<()> {
+	pub fn set_metadata(
+		token: TokenId,
+		name: Vec<u8>,
+		symbol: Vec<u8>,
+		decimals: u8,
+	) -> Result<()> {
 		build_dispatch(SET_METADATA)
-			.input::<(AssetId, Vec<u8>, Vec<u8>, u8)>()
+			.input::<(TokenId, Vec<u8>, Vec<u8>, u8)>()
 			.output::<Result<()>, true>()
 			.handle_error_code::<StatusCode>()
-			.call(&(id, name, symbol, decimals))
+			.call(&(token, name, symbol, decimals))
 	}
 
-	/// Clear the metadata for a token with a given asset ID.
+	/// Clear the metadata for a token.
 	///
 	/// # Parameters
-	/// - `id` - The ID of the asset.
+	/// - `token` - The token to update
 	#[inline]
-	pub fn clear_metadata(id: AssetId) -> Result<()> {
+	pub fn clear_metadata(token: TokenId) -> Result<()> {
 		build_dispatch(CLEAR_METADATA)
-			.input::<AssetId>()
+			.input::<TokenId>()
 			.output::<Result<()>, true>()
 			.handle_error_code::<StatusCode>()
-			.call(&(id))
+			.call(&(token))
 	}
 
-	/// Checks if token with a given asset ID exists.
+	/// Checks if a specified token exists.
 	///
 	/// # Parameters
-	/// - `id` - The ID of the asset.
+	/// - `token` - The token.
 	#[inline]
-	pub fn asset_exists(id: AssetId) -> Result<bool> {
-		build_read_state(ASSET_EXISTS)
-			.input::<AssetId>()
+	pub fn token_exists(token: TokenId) -> Result<bool> {
```

Comment:
As above, could just be exists()

Reply:
This is the api so if we change this we have to be sure. Since the standard uses `token_` in `token_decimals` etc, fits it not well with the rest?

Reply:
If there is no other exists I don't see the need to disambiguate. I prefer simple.

I assume the other token functions are named that way due to it being implemented on a contract and preventing clashing. 

But for the sake of finishing this, just leave it as is.